### PR TITLE
[terra-brand-footer] Removed Deprecated links Prop

### LINKED ISSUE #1122: [terra-brand-footer] Remove Deprecated links Prop
# Feature Request

## Description
Removed deprecated `links` prop in next MVB.

https://github.com/cerner/terra-framework/blob/e25da1097164dba3c32ecfaee2c45a2a2e751df0/packages/terra-brand-footer/src/BrandFooter.jsx#L11-L16

### PR DESCRIPTION
**Pull request by pranav300**
### Summary

This Pr removes the deprecated `links` prop

Closes #1122 

### Deployment Link

https://terra-framework-deployed-pr-1167.herokuapp.com/

### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-brand-footer/CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,8 @@ Changelog
 
 Unreleased
 ----------
+### Changed
```
Comment (emilyrohrbough) [inline packages/terra-brand-footer/CHANGELOG.md:6]: We need a MVB guide for this

Comment (pranav300) [reply packages/terra-brand-footer/CHANGELOG.md:6]: Updated here https://github.com/cerner/terra-framework/pull/1167/commits/06ebb6448986df1d498873f7164e9147d4b653c7.

File: packages/terra-brand-footer/src/terra-dev-site/doc/brand-footer/UpgradeGuide.2.doc.mdx [hunk 2]
```diff
@@ -3,6 +3,47 @@ import { Badge } from 'terra-brand-footer/package.json?dev-site-package';
...
++        ]}
+        />
+    );
+  }
+};
+```
+### Changes from version 2 to version 3
+#### Removed props
+* Removed `links`
+
```
Comment (emilyrohrbough) [inline packages/terra-brand-footer/src/terra-dev-site/doc/brand-footer/UpgradeGuide.2.doc.mdx:46]: Oops this was added 2x

File: packages/terra-brand-footer/src/BrandFooter.jsx [hunk 3]
```diff
@@ -105,10 +85,9 @@ const BrandFooter = ({
   customProps.className);
 
   let processedSections;
-  if (links.length > 0 && sections.length === 0) {
+  if (sections.length === 0) {
```
Comment (tbiethman) [inline packages/terra-brand-footer/src/BrandFooter.jsx:88]: This logic was executing previously if links were provided but sections were not. It wouldn't have executed in the case where no links and no sections were provided. Now that links are removed entirely, is this block necessary at all?

File: packages/terra-brand-footer/src/terra-dev-site/doc/brand-footer/UpgradeGuide.2.doc.mdx (hunk 2 above)
Comment (pranav300) [reply packages/terra-brand-footer/src/terra-dev-site/doc/brand-footer/UpgradeGuide.2.doc.mdx]: Updated here https://github.com/cerner/terra-framework/pull/1167/commits/b7d34c6ed6202c60788979859d48f35354d4d505.

File: packages/terra-brand-footer/src/BrandFooter.jsx (hunk 3 above)
Comment (pranav300) [reply packages/terra-brand-footer/src/BrandFooter.jsx]: Removed it here https://github.com/cerner/terra-framework/pull/1167/commits/b7d34c6ed6202c60788979859d48f35354d4d505.

File: packages/terra-brand-footer/src/BrandFooter.jsx [hunk 4]
```diff
@@ -104,20 +84,11 @@ const BrandFooter = ({
...
-    // Assign ids to use as keys
-    for (let i = 0; i < processedSections.length; i += 1) {
-      processedSections[i].id = i;
-    }
+  // Assign ids to use as keys
+  for (let i = 0; i < processedSections.length; i += 1) {
+    processedSections[i].id = i;
   }
 
   // The old links prop can't have section headers. Needed for vertical layout to insert padding to keep columns without headers aligned
```
Comment (emilyrohrbough) [inline packages/terra-brand-footer/src/BrandFooter.jsx:94]: Is this comment still applicable?

Comment (pranav300) [reply packages/terra-brand-footer/src/BrandFooter.jsx]: Oops!! missed it. Removed it here https://github.com/cerner/terra-framework/pull/1167/commits/64bf30125e4e5ebb2a7adc5693bbdc2b565fbb51

File: packages/terra-brand-footer/CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,9 @@ Changelog
...
```
Comment (mjhenkes) [inline packages/terra-brand-footer/CHANGELOG.md:6]:
```suggestion
### Breaking Changes
```

**Pull request merged**